File: c/src/hello.c
/**
 * Hello World WASI Component in C
 *
 * Exports a greet function that returns a greeting string.
 */

#include <stdio.h>
#include "hello_c.h"  // Generated WIT bindings

// Implementation of the greeter interface
void exports_hello_c_greeter_greet(hello_c_string_t *ret) {
    const char *greeting = "Hello wasm component world from C!";

    // Print to stdout (WASI handles this automatically)
    printf("%s\n", greeting);

    // Return the greeting string using the helper function
    hello_c_string_set(ret, greeting);
}

File: go/src/hello.go
// Simple Go hello world WASI component
//
// This demonstrates building a Go component using TinyGo and rules_wasm_component.
// The component exports a simple greeter interface.

package main

import "fmt"

// GreeterImpl implements the greeter interface exported by this component
type GreeterImpl struct{}

// Greet returns a greeting message
func (g *GreeterImpl) Greet() string {
	return "Hello from Go!"
}

func main() {
	// For reactor components, main() can be empty
	// The exports are handled through the WIT bindings
	fmt.Println("Hello from Go WASM component!")
}
